Project 4_KTs

```diff
@@ -0,0 +1,446 @@
+
+Example:
+
+Suppose you are given the URI of Borges: http://dbpedia.org/resource/Jorge_Luis_Borges. Your task is to write a SPARQL query that retrieves the titles of books he authored.
+
+Solution to SPARQL query:
+
+---
+PREFIX dbo: <http://dbpedia.org/ontology/>
+
+SELECT ?bookTitle
+WHERE {
+  <http://dbpedia.org/resource/Jorge_Luis_Borges> dbo:notableWork ?book .
+  ?book dbo:author <http://dbpedia.org/resource/Jorge_Luis_Borges> .
+  ?book dbo:name ?bookTitle .
+}
+ORDER BY ?bookTitle
+---
+
+Explanation:
+
+The first line defines the prefix for the DBpedia ontology namespace.
+The SELECT statement retrieves the title of the books authored by Borges using the WHERE clause.
+The WHERE clause specifies that the book must have Borges as the author using the dbo:author property, and must be a notable work using the dbo:notableWork property.
+The ?book variable refers to the book URI, and ?bookTitle refers to the book title.
+The results are sorted in ascending order by book title using the ORDER BY clause.
+
+
+Answer:
+
+Here are some results from this query:
+
+bookTitle
+------------------------------------------
+A Universal History of Iniquity
+Atlas
+Dreamtigers
+El aleph
+El hacedor
+Ficciones
+Historia universal de la infamia
+Inquisiciones
+La cuestión del otro
+La muerte y la brújula
+La biblioteca de Babel
+La lotería en Babilonia y otros relatos
+Nueve ensayos dantescos
+Otras inquisiciones
+Pierre Menard, autor del Quijote
+The Book of Imaginary Beings
+The Garden of Forking Paths
+The Maker
+The Secret Miracle
+The South
+The Theme of the Traitor and the Hero
+
+
+
+(3) level-2 kata (25 points/100 ponts)
+
+Title: Retrieve the titles, importance, and popularity of books authored by Jorge Luis Borges
+
+Description:
+
+You are not satisfied with simply listing out book titles and want to take your query to the next level. Write a SPARQL query that retrieves the titles, importance, and popularity of books authored by Jorge Luis Borges. You will be given the URI of Borges, and your task is to extract the titles of the books he authored, along with information on why they are important and why they are popular, using the DBpedia ontology.
+
+Example:
+
+Suppose you are given the URI of Borges: http://dbpedia.org/resource/Jorge_Luis_Borges. Your task is to write a SPARQL query that retrieves the titles, importance, and popularity of books he authored.
+
+
+Solution to SPARQL query:
+
+---
+PREFIX dbo: <http://dbpedia.org/ontology/>
+PREFIX dbp: <http://dbpedia.org/property/>
+
+SELECT DISTINCT ?bookTitle ?importance ?popularity
+WHERE {
+  <http://dbpedia.org/resource/Jorge_Luis_Borges> dbo:notableWork ?book .
+  ?book dbo:author <http://dbpedia.org/resource/Jorge_Luis_Borges> .
+  ?book dbo:name ?bookTitle .
+  OPTIONAL { ?book dbp:importance ?importance } .
+  OPTIONAL { ?book dbp:popularity ?popularity } .
+}
+ORDER BY ?bookTitle
+---
```

Comment:
Again this seems like a Kata-6 difficulty, at most.

Kata 2's are expert level familiarity and competence. To make this a Kata-2, I would expect a _stack_ of conceptually challenging tasks, such as asserting a new triple, (CONSTRUCT, INSERT, CONCAT, or the like), putting some limits that require laser-like precision, AND some unavoidably complicated syntax. E.g., a filter for a string or a negation that could easily violate open-world assumption, or to relate the works under a new heading given some shared attribute that is hard to string together. Chains of reasoning implying further triples, BIND ... AS, and so on.